### PR TITLE
es6 + Bearer auth

### PR DESCRIPTION
**Pull request by wookiehangover**
#### What does it do?

Converts the client code to leverage es6 using [6to5](http://6to5.org/) to compile down to es5. This fits well with our other projects that rely on browserify transform steps for jsx, and lets us use the new stuff now.

Also adds support for bearer auth to go along with out new Oauth2 workflow. There's a new signature for creating a functioning client with just an access token:

```javascript
var client = sprintlyData.createClient({ token: 'abc123' });
```

NB that you must obtain an access token with a server-side oauth workflow.

#### Where should the reviewer start?

Most of these changes are superficial conversions of module imports.

The bearer auth header is in `lib/support/bearer-auth.js`

#### Any background context you want to provide?

This has been a long running branch that I got the motivation to finish out when needing to add Bearer auth.

It's worth noting that I also bump the release version to `1.0.0`. Having been in `0.x.0` releases and not officially published to npm while we were getting our sea-legs, I've decided that with the addition of official client support

Let the Age of OAuth clients begin!
url

### CODE REVIEW

File: lib/items/index.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-var _ = require('lodash');
-var Backbone = require('../support/backbone');
-var config = require('../config');
-var deps = require('ampersand-dependency-mixin');
-var Item = require('./item');
-var QueryConfig = require('./query-config');
+import _ from 'lodash';
+import Backbone from '../support/backbone';
```
Comment (justinabrahms) [inline lib/items/index.js]: Looks like we only use `Collection` here?

**Comment by justinabrahms** [comment]
Looks good to me, but I'd like @florapdx to have a look to familiarize herself with es6 syntax in a relatively small repo.

File: lib/items/index.js (hunk 1 above)
Comment (wookiehangover) [inline lib/items/index.js]: Good catch

**Comment by wookiehangover** [comment]
@florapdx gotten a chance to peek at this one yet? Once landed I'm going to publish to npm, since we announced that in a blog post today :smiley_cat:

**Comment by florapdx** [comment]
Just sitting down to it now, @wookiehangover :)

**Comment by florapdx** [comment]
:+1: Thanks for giving me a chance to see this before merging!

**Pull request merged**